packages/core-ethereum/src/types/accountId.ts

```diff
@@ -1,28 +1,26 @@
 import Web3 from 'web3'
 import type { Types } from '@hoprnet/hopr-core-connector-interface'
 import { ADDRESS_LENGTH } from '../constants'
-import { Uint8ArrayE } from './extended'
+import { u8aToHex, u8aEquals } from '@hoprnet/hopr-utils'
 
-class AccountId extends Uint8ArrayE implements Types.AccountId {
-  slice(begin = 0, end = AccountId.SIZE): Uint8Array {
-    return this.subarray(begin, end)
-  }
-
-  subarray(begin = 0, end = AccountId.SIZE): Uint8Array {
-    return new Uint8Array(this.buffer, begin + this.byteOffset, end - begin)
-  }
+class Address implements Types.Address {
+  constructor(private id: Uint8Array) {}
 
   static get SIZE(): number {
     return ADDRESS_LENGTH
   }
 
-  get NAME(): string {
-    return 'AccountId'
+  serialize() {
+    return this.id
   }
 
   toHex(): string {
-    return Web3.utils.toChecksumAddress(super.toHex())
+    return Web3.utils.toChecksumAddress(u8aToHex(this.id, false))
```

Comment:
Not sure whether we should keep Ethereum checksums here.

0xcA is not the same as 0xca and .toHex() can be misunderstood as "return a hex string like 0x123456789abcdef"

Reply:
we should use checksum whenever possible

Reply:
This is a good point, why we had `.toChecksumAddress` in the first place?

Reply:
ah I misunderstood @robertkiel, let's rename `toHex` to `toChecksumAddress`, we want to convert address to it's string representation, it should be checksummed